x-pack/test/apm_api_integration/tests/alerts/preview_chart_error_rate.spec.ts

```diff
@@ -48,243 +49,244 @@ export default function ApiTest({ getService }: FtrProviderContext) {
     });
   });
 
-  registry.when(`with data loaded`, { config: 'basic', archives: [archiveName] }, () => {
-    it('transaction_error_rate (with data)', async () => {
-      const options = getOptions();
-      const response = await apmApiClient.readUser({
-        endpoint: 'GET /internal/apm/rule_types/transaction_error_rate/chart_preview',
-        ...options,
+  registry.when(`with data loaded`, { config: 'basic', archives: [] }, () => {
+    describe('errors distribution', () => {
+      before(async () => {
+        await generateData({ serviceName, start, end, synthtraceEsClient });
       });
 
-      expect(response.status).to.be(200);
-      expect(
-        response.body.errorRateChartPreview.some(
-          (item: { name: string; data: Array<{ x: number; y: number | null }> }) =>
-            item.data.some((coordinate) => coordinate.x && coordinate.y)
-        )
-      ).to.equal(true);
-    });
+      after(() => synthtraceEsClient.clean());
 
-    it('transaction_error_rate with transaction name', async () => {
-      const options = {
-        params: {
-          query: {
-            start,
-            end,
-            serviceName: 'opbeans-java',
-            transactionName: 'APIRestController#product',
-            transactionType: 'request',
-            environment: 'ENVIRONMENT_ALL',
-            interval: '5m',
-          },
-        },
-      };
+      it('transaction_error_rate (with data)', async () => {
+        const options = getOptions();
+        const response = await apmApiClient.readUser({
+          endpoint: 'GET /internal/apm/rule_types/transaction_error_rate/chart_preview',
+          ...options,
+        });
 
-      const response = await apmApiClient.readUser({
-        endpoint: 'GET /internal/apm/rule_types/transaction_error_rate/chart_preview',
-        ...options,
-      });
+        expect(response.status).to.be(200);
+        expect(
+          response.body.errorRateChartPreview.some(
+            (item: { name: string; data: Array<{ x: number; y: number | null }> }) =>
+              item.data.some((coordinate) => coordinate.x && coordinate.y)
+          )
+        ).to.equal(true);
 
-      expect(response.status).to.be(200);
-      expect(response.body.errorRateChartPreview[0].data[0]).to.eql({
-        x: 1627973700000,
-        y: 100,
+        expect(response.body.errorRateChartPreview[0].name).to.eql('synth-go_production_request');
+        expect(response.body.errorRateChartPreview[0].data[1]).to.eql({
+          x: 1609459500000,
+          y: 37.5,
```

Comment:
Is 37.5% the expected failure rate?

Reply:
Yes. This is the combined failure rate of both transactions.
Failure rate of `GET /apple` is 25 and `GET /banana` is 50. So the combined failure rate for `synth-go` service is `(50 + 25) / 2 = 37.5`.

Reply:
I think it doesn't need to be defined in data generation. This value is calculated based on the available data.